src/diffusers/models/autoencoder_kl.py

```diff
@@ -249,7 +249,21 @@ def set_default_attn_processor(self):
         self.set_attn_processor(processor, _remove_lora=True)
 
     @apply_forward_hook
-    def encode(self, x: torch.FloatTensor, return_dict: bool = True) -> AutoencoderKLOutput:
+    def encode(
+        self, x: torch.FloatTensor, return_dict: bool = True
+    ) -> Union[AutoencoderKLOutput, tuple[DiagonalGaussianDistribution]]:
```

Comment:
```suggestion
    ) -> Union[AutoencoderKLOutput, Tuple[DiagonalGaussianDistribution]]:
```

to support Python 3.8 syntax